### PR TITLE
Minor crt1 optimizations

### PR DESCRIPTION
**Pull request by kripken**
* Avoid stack allocation in `_start`, as the stack push/pop cannot be optimized away, and are wasteful without a main.
 * Don't export `__wasm_call_ctors` as we call it from `_start` on the inside.
 * Build crt1 with optimizations instead of `-O0`.

### CODE REVIEW

File: tools/shared.py [hunk 1]
```diff
@@ -1903,9 +1903,14 @@ def link_lld(args, target, opts=[], lto_level=0):
     if Settings.LINKABLE:
       cmd.append('--export-all')
 
+    # in standalone mode, crt1 will call the constructors from inside the wasm
+    if not Settings.STANDALONE_WASM:
```
Comment (sbc100) [inline tools/shared.py]: Except when there is no entry point we do want to export it.  Perhaps this should be guarded by `use_start_function` below?

Also might as will do this on one line: `cmd += [  '--export', '__wasm_call_ctors']`

File: tools/shared.py [hunk 2]
```diff
@@ -1903,9 +1903,14 @@ def link_lld(args, target, opts=[], lto_level=0):
...
+        '--export',
+        '__wasm_call_ctors',
+      ]
+
     cmd += [
-      '--export',
-      '__wasm_call_ctors',
       '--export',
       '__data_end'
     ]
```
Comment (sbc100) [inline tools/shared.py]: This can be folded onto single line too, now that its only one thing to export.

File: tools/shared.py (hunk 1 above)
Comment (kripken) [reply tools/shared.py]: I'm unclear on what "no entry point" means. When does that happen?

Comment (kripken) [reply tools/shared.py]: (and do we not have tests for it?)

Comment (sbc100) [reply tools/shared.py]: OK I think you right.   I though we were avoiding the _start function when building a STANDALONE_WASM without main, but actaully we still link _start in that case, its just weakly linked to main so it OK for main to be missing.

This change lgtm then, with cleanup

**Pull request merged**